compiler/noirc_driver/src/abi_gen.rs

```diff
@@ -82,6 +81,14 @@ pub(super) fn abi_type_from_hir_type(context: &Context, typ: &Type) -> AbiType {
             AbiType::Array { length, typ: Box::new(abi_type_from_hir_type(context, typ)) }
         }
         Type::Integer(sign, bit_width) => {
+            // TODO: combine Bool, etc in ABI
```

Comment:
This doesn't seem wise as we want to be able to differentiate between these types when accepting user input.